docs/tutorials/prompt_engineering/chaining_based/demonstration_ensembling.ipynb

```diff
@@ -0,0 +1,208 @@
+    "\n",
+    "<div class=\"admonition tip\">\n",
+    "<p class=\"admonition-title\">Additional Real-World Applications</p>\n",
+    "<ul>\n",
+    "<li><b>Code Generation</b>: Break down complex programming tasks into smaller, manageable steps.</li>\n",
+    "<li><b>Data Analysis</b>: Decompose complex data analysis queries into a series of data manipulation and calculation steps.</li>\n",
+    "<li><b>Natural Language Processing</b>: Break down complex NLP tasks like sentiment analysis or named entity recognition into subtasks.</li>\n",
+    "<li><b>Automated Reasoning</b>: Solve complex logical or mathematical problems by breaking them into simpler, solvable steps.</li>\n",
+    "<li><b>Task Planning</b>: Create detailed, step-by-step plans for complex projects or processes.</li>\n",
+    "</ul>\n",
+    "</div>\n",
+    "\n",
+    "## Implementation\n",
+    "\n",
+    "Let's implement the Demonstration Ensembling technique using Mirascope:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "4fa42cb9047d3c03",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-10-01T01:34:48.511560Z",
+     "start_time": "2024-10-01T01:34:40.601245Z"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Here's an aggregated notice highlighting the importance of attending the company's social events:\n",
+      "\n",
+      "---\n",
+      "\n",
+      "**Notice: Importance of Attending Company Social Events**\n",
+      "\n",
+      "Dear Team,\n",
+      "\n",
+      "We would like to emphasize the significance of participating in our upcoming company social events. These gatherings provide valuable opportunities to relax, unwind, and strengthen relationships among colleagues. They play a vital role in fostering teamwork, enhancing collaboration, and building a strong company culture.\n",
+      "\n",
+      "Engaging with each other outside of the typical work environment promotes open communication, encourages creativity, and strengthens our sense of community within the organization. Your participation not only enriches your own experience but also contributes to creating a positive and inclusive work atmosphere.\n",
+      "\n",
+      "We encourage everyone to take advantage of these occasions to connect with teammates, share ideas, and create lasting memories. Thank you for your continued engagement and support. We look forward to seeing you at the next social event!\n",
+      "\n",
+      "Best regards,  \n",
+      "[Your Company Name] Management Team  \n",
+      "\n",
+      "--- \n",
+      "\n",
+      "This notice captures the various elements emphasized in the different responses while maintaining a cohesive and professional tone.\n"
+     ]
+    }
+   ],
+   "source": [
+    "import asyncio\n",
+    "import random\n",
+    "\n",
+    "from mirascope.core import openai, prompt_template\n",
+    "\n",
+    "qa_examples = [\n",
```

Comment:
we should structure these examples better, maybe using base models or typed dicts?